integration_tests/broker_test.go

```diff
@@ -0,0 +1,199 @@
+package integration_test
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"strconv"
+
+	"github.com/alphagov/paas-compose-broker/catalog"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	uuid "github.com/satori/go.uuid"
+)
+
+var (
+	responseRecorder  *httptest.ResponseRecorder
+	instanceID        string
+	serviceID         string
+	planID            string
+	bindingID         string
+	paramJSON         string
+	acceptsIncomplete bool
+)
+
+var _ = Describe("Broker", func() {
+
+	BeforeEach(func() {
+		responseRecorder = httptest.NewRecorder()
+		instanceID = uuid.NewV4().String()
+		serviceID = "Service-1"
+		planID = "Plan-1"
+		bindingID = uuid.NewV4().String()
+		paramJSON = "{}"
+		acceptsIncomplete = true
+	})
+
+	It("serves the catalog endpoint", func() {
+		req := newRequest("GET", "/v2/catalog", nil)
+		resp := serveRequest(responseRecorder, req)
+		Expect(responseRecorder.Code).To(Equal(http.StatusOK))
+		body := readResponseBody(resp)
+		returnedCatalog, err := catalog.New(bytes.NewBuffer(body))
```

Comment:
I missed this on the first pass, but it's not great to use the `catalog` in the tests here because it ends up being somewhat circular. This means that if we're JSON encoding the catalog in a way that's different to the spec the tests won't catch it. Using a `map[string]interface{}` means that we can assert on the specific contents of the JSON.

Another small point- `resp.Body` is already an `io.Reader`, so there's no need to read it into a byte slice and then wrap that in a `bytes.Buffer`, we can pass `resp.Body` directly to `catalog.New`.